### PR TITLE
Adding AML 301 and updates to 201 and 202

### PR DESCRIPTION
**Pull request by ryhud**
@grayzu @dapolloxp @djr1991 @denniseik 

Adding AML 301 sample. Updating AML 201 with bastion and DSVM. Also fixing issue with 201 where destroy fails by adding a dependency to the training subnet for the workspace.

### CODE REVIEW

File: quickstart/301-machine-learning-hub-spoke-secure/main.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+terraform {
+  required_version = ">=0.15.0"
```
Comment (grayzu) [inline quickstart/301-machine-learning-hub-spoke-secure/main.tf]: Why are you using 0.15 instead of 1.0?

**Review by grayzu — COMMENTED** [review]
LGTM with one minor comment.

Also, it looks like you may have already but to be sure, please run terraform fmt to ensure that all of the files are formatted correctly.

File: quickstart/301-machine-learning-hub-spoke-secure/main.tf (hunk 1 above)
Comment (ryhud) [reply quickstart/301-machine-learning-hub-spoke-secure/main.tf]: @grayzu, updated the versions to 1.0 and ran fmt

**Review by grayzu — CHANGES_REQUESTED** [review]
Please remove .gitignore file and I will go ahead and merge.

This looks fantastic, btw!! Thank you so much for the contribution.

**Comment by ryhud** [comment]
@grayzu DONE

**Pull request merged**